ksz8863/components/ksz8863-cmd/ksz8863_cmd.c

```diff
@@ -0,0 +1,331 @@
+{
+    host_handle = h_handle;
+    port_handles[0] = p1_handle;
+    port_handles[1] = p2_handle;
+
+    s_reset_args.type = arg_str0(NULL, NULL, "<reset type>", "Specify either 'soft' or 'hard' to perform reset");
+    s_reset_args.end = arg_end(1);
+
+    s_set_args.port = arg_int0(NULL, "port", "<port_num>", "Port in which the change will be made");
+    s_set_args.property = arg_str0(NULL, NULL, "<property>", "Property to be set");
+    s_set_args.value = arg_int0(NULL, NULL, "<value>", "New value for the property");
+    s_set_args.end = arg_end(3);
+
+    s_start_args.port = arg_int0(NULL, "port", "<port_num>", "Port in which the change will be made");
+    s_start_args.status = arg_str0(NULL, NULL, "<status>", "Up to enable port and down to disable it"),
+    s_start_args.end = arg_end(1);
+
+    const esp_console_cmd_t switch_reset_cmd = {
+        .command = "swreset",
+        .help = "Perofrm either soft or hard reset of the switch",
+        .hint = NULL,
+        .func = &cmd_switch_reset,
+        .argtable = &s_reset_args
+    };
+    const esp_console_cmd_t switch_set_cmd = {
+        .command = "set",
+        .help = "Set control value for given property",
+        .hint = NULL,
+        .func = &cmd_switch_set,
+        .argtable = &s_set_args
+    };
+    const esp_console_cmd_t switch_start_cmd = {
+        .command = "bring",
+        .help = "Bring selected port up or down",
+        .hint = NULL,
+        .func = &cmd_switch_bring,
+        .argtable = &s_start_args
+    };
+
+    ESP_ERROR_CHECK(esp_console_cmd_register(&switch_reset_cmd));
+    ESP_ERROR_CHECK(esp_console_cmd_register(&switch_set_cmd));
+    ESP_ERROR_CHECK(esp_console_cmd_register(&switch_start_cmd));
+}*/
+
+static int cmd_switch(int argc, char **argv)
+{
+    int nerrors = arg_parse(argc, argv, (void **) &s_switch_args);
+    if (nerrors != 0) {
+        arg_print_errors(stderr, s_switch_args.end, argv[0]);
+        return 1;
+    }
+
+    int port = s_switch_args.port->ival[0];
+    char *action = s_switch_args.action->sval[0];
+    char *parameter = s_switch_args.parameter->sval[0];
+    char *value = s_switch_args.value->sval[0];
+
+    if (strcmp(action, "set") == 0) {
+        // set parameters such as tx/rx status
+        if (strcmp(parameter, "rx") == 0) {
+            bool new_value = atoi(s_switch_args.value->sval[0]) == 1;
+            esp_eth_ioctl(port_handles[port - 1], KSZ8863_ETH_CMD_S_RX_EN, &new_value);
+        } else if (strcmp(parameter, "tx") == 0) {
+            bool new_value = atoi(s_switch_args.value->sval[0]) == 1;
+            esp_eth_ioctl(port_handles[port - 1], KSZ8863_ETH_CMD_S_TX_EN, &new_value);
+        } else if (strcmp(parameter, "tailtag") == 0) {
+            bool new_value = atoi(s_switch_args.value->sval[0]) == 1;
+            esp_eth_ioctl(port_handles[port - 1], KSZ8863_ETH_CMD_S_TAIL_TAG, &new_value);
```

Comment:
Is this option safe to be exposed to user via console? I mean, what will happen when switch is initialized in tail tag mode and then it's disabled?